src/time-picker/time-picker.jsx

```diff
@@ -178,6 +190,18 @@ const TimePicker = React.createClass({
     if (this.props.onTouchTap) this.props.onTouchTap(e);
   },
 
+  _isControlled() {
+    return this.props.hasOwnProperty('value');
```

Comment:
What about setting `null` as the default value for `value` and simplify the check with `this.props.value !== null`. I'm even wondering if we need this `_isControlled` method.